src/librustc_save_analysis/dump_visitor.rs

```diff
@@ -115,15 +117,18 @@ impl<'l, 'tcx: 'l, 'll, D: Dump + 'll> DumpVisitor<'l, 'tcx, 'll, D> {
         where F: FnOnce(&mut DumpVisitor<'l, 'tcx, 'll, D>)
     {
         let item_def_id = self.tcx.hir.local_def_id(item_id);
+        let old_item_cx = mem::replace(&mut self.save_ctxt.item_cx,
+                                       ItemCtxt::new(self.tcx, item_def_id));
```

Comment:
Hah I didn't expect this. I'm not sure we should have this at all. That is, `ItemCtxt` is just a `(TyCtxt, DefId)` pair at this point IIRC, and the `item_def_id` seems to be the important thing to keep?